test/unit/player_test.rb

```diff
@@ -0,0 +1,55 @@
+require 'test_helper'
+
+class PlayerTest < ActiveSupport::TestCase
+  # test "the truth" do
+  #   assert true
+  # end
+
+  test "validates for name, level, armor class, and passive perception" do
+    player = Player.new
+    assert !player.save
+    player.name = "bob"
+    assert !player.save
+    player.level = 3
+    assert !player.save
```

Comment:
I've also been taught to use .save! instead of save to ensure that an exception is raised instead of just returning false.